mmv1/templates/terraform/state_migrations/certificate_manager_dns_authorization.go.erb

```diff
@@ -0,0 +1,91 @@
+func ResourceCertificateManagerDnsAuthorizationUpgradeV0(_ context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	log.Printf("[DEBUG] Attributes before migration: %#v", rawState)
```

Comment:
I noticed that this log is not printed in while running the migration test, when it will be used? 

Also what can I do to mimic a customer that has a configuration in an old version and then upgrades it? I didn't know how to define the resource to be in an older version. 

I had an old journey with a similar change and you suggested some solutions for testing: https://github.com/GoogleCloudPlatform/magic-modules/pull/7661#issuecomment-1503888510, I would appreciate  if you could elaborate more on both of the mentioned solutions.

Reply:
>I noticed that this log is not printed in while running the migration test, when it will be used?

I ran your `TestAccCertificateManagerDnsAuthorization_migration` test locally with the environment variable TF_LOG set to "DEBUG" to make debug logs visible and I also didn't see the log. I believe this shows that the test isn't using the local code from `mynewprovider` - see other comment for suggested changes!

>Also what can I do to mimic a customer that has a configuration in an old version and then upgrades it? I didn't know how to define the resource to be in an older version.

You can achieve this with the `TestAccCertificateManagerDnsAuthorization_migration` test (following the fix in my other comment). In that test what's happening is:

1. the resource is being provisioned with the old version of the provider + the first config
2. test verifies ability to import the resource using the old version of the provider
3. the resource is being updated using the new version of the provider + the second config with updated description
4. test verifies ability to import the resource using the new version of the provider

You could add further test steps if they were needed. For example you could insert a step between 2 and 3 of the steps above where you use the first config with the new provider and check that the plan is empty (the snippet below has `ExpectNonEmptyPlan: false,` implied). This would mimic a user just upgrading their provider version and not changing their config:

```
			{
				Config:            dnsAuthorizationResourceConfig(name),
				ProviderFactories: newVersion,
				PlanOnly:          true,
			},
```